example/ida/utils.py

```diff
@@ -72,22 +73,29 @@ class TranslatorIDA(Translator):
     # Implemented language
     __LANG__ = "ida_w_color"
 
-    def __init__(self, regs_ids=None, **kwargs):
+    def __init__(self, symbol_pool=None, **kwargs):
         super(TranslatorIDA, self).__init__(**kwargs)
-        if regs_ids is None:
-            regs_ids = {}
-        self.regs_ids = regs_ids
+        self.symbol_pool = symbol_pool
 
-    def str_protected_child(self, child, parent):
-        return ("(%s)" % self.from_expr(child)) if m2_expr.should_parenthesize_child(child, parent) else self.from_expr(child)
+    def str_protected_child(self, child, parent, **kwargs):
+        return ("(%s)" % (
+            self.from_expr(child, **kwargs)) if m2_expr.should_parenthesize_child(child, parent)
+                else self.from_expr(child, **kwargs)
+        )
 
     def from_ExprInt(self, expr):
         return idaapi.COLSTR(str(expr), idaapi.SCOLOR_NUMBER)
 
     def from_ExprId(self, expr):
-        out = str(expr)
-        if expr in self.regs_ids:
-            out = idaapi.COLSTR(out, idaapi.SCOLOR_REG)
+        out = idaapi.COLSTR(str(expr), idaapi.SCOLOR_REG)
```

Comment:
`ExprId` will now always be in this color. In the previous version, only the registers (through a whitelist) were concerned. Is it on purpose?

Reply:
yes: I think it's interesting to have all IDs colored. But it's completly arbitrary.
If you feel the UI may be enhanced with only colored regs, tell me

Reply:
Fair enough